qoqo/src/noise_models/overrotation.rs

```diff
@@ -0,0 +1,408 @@
+    ///
+    /// Raises:
+    ///     ValueError: Cannot serialize SingleQubitOverrotationDescription.
+    ///
+    pub fn to_json(&self) -> PyResult<String> {
+        let noise_descp = self.internal.clone();
+        let serialized = serde_json::to_string(&noise_descp).map_err(|_| {
+            pyo3::exceptions::PyValueError::new_err(
+                "Cannot serialize single qubit overrotation description to json.",
+            )
+        })?;
+        Ok(serialized)
+    }
+
+    #[cfg(feature = "json_schema")]
+    /// Return the minimum version of qoqo that supports this object.
+    ///
+    /// Returns:
+    ///     str: The minimum version of the qoqo library to deserialize this object.
+    pub fn min_supported_version(&self) -> String {
+        let min_version: (u32, u32, u32) =
+            SingleQubitOverrotationDescription::minimum_supported_roqoqo_version(
+                &self.internal.clone(),
+            );
+        format!("{}.{}.{}", min_version.0, min_version.1, min_version.2)
+    }
+
+    /// Return the __richcmp__ magic method to perform rich comparison operations on mixed system.
+    ///
+    /// Args:
+    ///     other: The object to compare self to.
+    ///     op: Whether they should be equal or not.
+    ///
+    /// Returns:
+    ///     bool
+    ///
+    /// Raises:
+    ///     NotImplementedError: Other comparison not implemented.
+    ///
+    fn __richcmp__(&self, other: Py<PyAny>, op: pyo3::class::basic::CompareOp) -> PyResult<bool> {
+        let other = SingleQubitOverrotationDescriptionWrapper::from_pyany(other);
+
+        match op {
+            pyo3::class::basic::CompareOp::Eq => match other {
+                Ok(osystem) => Ok(self.internal.clone() == osystem),
+                _ => Ok(false),
+            },
+            pyo3::class::basic::CompareOp::Ne => match other {
+                Ok(osystem) => Ok(self.internal.clone() != osystem),
+                _ => Ok(true),
+            },
+            _ => Err(pyo3::exceptions::PyNotImplementedError::new_err(
+                "Other comparison not implemented",
+            )),
+        }
+    }
+}
+
+/// Single qubit overrotation noise model on gate.
+///
+/// Adds a rotatation gate with a randomly distributed rotation angle after specified gates in a quantum circuit.
+/// Example:
+///
+/// ```
+/// from qoqo.noise_models import SingleQubitOverrotationDescription
+/// from qoqo.noise_models import SingleQubitOverrotationOnGate
+/// gate = "RotateX"
+/// theta_mean = 0.0
+/// theta_std = 1.0
+/// noise_desc = SingleQubitOverrotationDescription(gate, theta_mean, theta_std)
+///
+/// noise = SingleQubitOverrotationOnGate();
+/// circuit_gate_with_noise = "RotateZ";
+/// qubit = 0;
+/// noise.set_single_qubit_overrotations(circuit_gate_with_noise, qubit, noise_desc);
+/// ```
+#[pyclass(frozen, name = "SingleQubitOverrotationOnGate")]
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct SingleQubitOverrotationOnGateWrapper {
```

Comment:
This struct seems to be missing some functions, such as `__richcmp__`, `__copy__` and `__deepcopy__` amongst others

Reply:
The `#[noise_model_wrapper]` macros is used for this below which contains all these functions.